### ISSUE TITLE
Please add the default compiler as a build argument

### ISSUE THREAD

**Issue by yurivict** [Theano/Theano#6584]
Currently it requires cxx=c++ in ```~/.theanorc```, which is inconvenient. It's easier if the right default is baked into the package.

**Comment by lamblin** [comment]
Currently, only `g++` is supported on most platforms, except for MacOS where we only support `clang++`.
This is what we use for the default in the package, see for instance here: https://github.com/Theano/Theano/blob/master/theano/configdefaults.py#L496

Do you have examples where `cxx=c++` works, but `g++` is not present on the system?

**Comment by yurivict** [comment]
No, I asked for a way to set a default it during build, so that there will be no need to create ```~/.theanorc```.

Additionally, FreeBSD should also use clang++ in Theano, because clang is a default compiler on FreeBSD.

**Comment by lamblin** [comment]
I'm not sure I understand what you mean by "build".
There is no real "build" or "packaging" phase, apart from the `setup.py`.
Theano does not have any mechanism to set global defaults at the moment, and the only way to set options are config files (`.theanorc` or wherever `$THEANORC` points to), `$THEANO_FLAGS`, and the code itself.
Since MILA stopped actively developing Theano, we will not add a new mechanism, but if you want to add a special case for FreeBSD in `configdefaults.py`, please submit a PR.

**Comment by yurivict** [comment]
I mean to have a compile-time default for cxx in ~/.theanorc, such that this file isn't required.

**Comment by lamblin** [comment]
There is no "compile-time" when installing Theano, it's just a Python package, and everything is compiled on-demand at execution time.
There is no build phase where options can be provided between the download of the source code and the time the package is imported.
